govcd/entity_test.go

```diff
@@ -219,6 +218,5 @@ func (vcd *TestVCD) testFinderGetGenericEntity(def getterTestDefinition, check *
 	ge, err = def.getByNameOrId(invalidEntityId, false)
 	entity8 := ge.(genericEntity)
 	check.Assert(err, NotNil)
-	check.Assert(IsNotFound(err), Equals, true)
```

Comment:
Is there a specific reason to remove this check (and the above one)? I think other tests using this `testFinderGetGenericEntity` rely and satisfy this depedency. Aren't we "relieving" this for all entities?

Reply:
It questionable. My decision was to remove cause only specific one object can use that, all other entitities can't comply as Get*ByHref is direct call and would return 403 and not found error.

Reply:
This check **must not** be removed.
`GetDisk*` functions will not use it. And if other entities fail this check, it means that they were not well defined.

The failure here must return a ErrEntityNotFound because the code should not reach `Get*ByHREF`. If it does, the implementation should be reviewed.

Reply:
GetDisk don't use this generic method, but media has same approach. Difference is that no all GetHref methods works in the way that fetch all elements and filter out by ==href. Normally you just call API ir return error if Href is wrong.

Reply:
Please show me where you think it is inappropriate and I will see if the test can be improved. But until further agreement, this test works as defined and we cannot change it.

Reply:
GetExternalNetworkById vs GetMediaById

Reply:
GetMediaById can be improved, as commented in its PR. 
If there is still a problem with it, it will be solved in such place. As for the current PR, this check should be reinstated.

Reply:
reverted